pkg/api/pod/util.go

```diff
@@ -678,6 +678,58 @@ func dropDisabledFields(
 	dropPodLifecycleSleepAction(podSpec, oldPodSpec)
 	dropImageVolumes(podSpec, oldPodSpec)
 	dropSELinuxChangePolicy(podSpec, oldPodSpec)
+	dropContainerStopSignals(podSpec, oldPodSpec)
+}
+
+func dropContainerStopSignals(podSpec, oldPodSpec *api.PodSpec) {
+	if utilfeature.DefaultFeatureGate.Enabled(features.ContainerStopSignals) || containerStopSignalsInUse(oldPodSpec) {
+		return
+	}
+
+	adjustLifecycle := func(lifecycle *api.Lifecycle) {
+		if lifecycle.StopSignal != nil || *lifecycle == (api.Lifecycle{}) {
+			lifecycle.StopSignal = nil
+		}
+	}
+
+	for i := range podSpec.Containers {
```

Comment:
I think you just reinvented `VisitContainers()` - can you use that instead?

Reply:
Done in https://github.com/kubernetes/kubernetes/pull/130556/commits/a857f239816bf42ecbf58033ae7d1076be85de34. This was how it was done in the `dropPodLifecycleSleepAction`, we can probably refactor that function too.